packages/nx/src/daemon/server/project-graph-incremental-recomputation.ts

```diff
@@ -117,8 +117,36 @@ export async function getCachedSerializedProjectGraphPromise(): Promise<Serializ
       );
     }
 
+    const errors = result.error
+      ? result.error instanceof DaemonProjectGraphError
+        ? result.error.errors
+        : [result.error]
+      : [];
+
+    // Always write the daemon's current graph to disk to ensure disk cache
+    // stays in sync with the daemon's in-memory cache. This prevents issues
+    // where a non-daemon process writes a stale/errored cache that never
+    // gets overwritten by the daemon's valid graph.
+    if (
+      result.projectGraph &&
+      result.projectFileMapCache &&
+      result.sourceMaps
+    ) {
+      writeCache(
+        result.projectFileMapCache,
+        result.projectGraph,
+        result.sourceMaps,
+        errors
+      );
+    }
+
     return result;
   } catch (e) {
+    if (e instanceof DaemonProjectGraphError) {
```

Comment:
I don't think this is ever possible.. is it? I'd remove this. writing to the cache above already handles the cases we're seeing I htink.